/*
 * Copyright (c) 2020 Huawei Technologies Co.,Ltd.
 *
 * openGauss is licensed under Mulan PSL v2.
 * You can use this software according to the terms and conditions of the Mulan PSL v2.
 * You may obtain a copy of Mulan PSL v2 at:
 *
 *          http://license.coscl.org.cn/MulanPSL2
 *
 * THIS SOFTWARE IS PROVIDED ON AN "AS IS" BASIS, WITHOUT WARRANTIES OF ANY KIND,
 * EITHER EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO NON-INFRINGEMENT,
 * MERCHANTABILITY OR FIT FOR A PARTICULAR PURPOSE.
 * See the Mulan PSL v2 for more details.
 *---------------------------------------------------------------------------------------
 *
 * IDENTIFICATION
 *        src/gausskernel/security/keymanagement/TDEKeysRecords.proto
 *
 * ---------------------------------------------------------------------------------------
 */
syntax = "proto2";
message TDEKeysRecords{
  enum DEK_STATUS{
        ALIVED=1;
        DEAD=0;
        INVALID=-1;
  }
  required bytes table_id = 1;
  required bytes database_id = 2;
  required DEK_STATUS DEK_status = 3 [default = DEAD];
  required bytes CEK_name = 4;
  required bytes CEK_version = 5;
  required bytes CEK_IV = 6;
  required bytes encrypted_sample_string = 7;
  required bytes DEK_cipher = 8;
  repeated fixed32 DEK_IV = 10;  
  enum TDE_ALGO{
    AES_CTR_128=1;
    SM4_CTR_128=2;
  }
  required TDE_ALGO Algo = 9 [default = AES_CTR_128];
}
